pages/instructorFileBrowser/instructorFileBrowser.js

```diff
@@ -297,47 +297,59 @@ function browseDirectory(file_browser, callback) {
 function browseFile(file_browser, callback) {
   async.waterfall(
     [
-      (callback) => {
-        fs.readFile(file_browser.paths.workingPath, (err, contents) => {
-          if (ERR(err, callback)) return;
-          callback(null, contents);
-        });
-      },
-      (contents, callback) => {
-        isBinaryFile(contents).then(
-          (result) => {
-            debug(`isBinaryFile: ${result}`);
-            file_browser.isBinary = result;
-            if (result) {
-              util.callbackify(async () => {
-                const type = await FileType.fromBuffer(contents);
-                if (type) {
-                  debug(`file type:\n ext = ${type.ext}\n mime = ${type.mime}`);
-                  if (type.mime.startsWith('image')) {
-                    file_browser.isImage = true;
-                  } else if (type.mime === 'application/pdf') {
-                    file_browser.isPDF = true;
-                  }
-                } else {
-                  debug(`could not get file type`);
-                }
-              })(callback);
-            } else {
-              debug(`found a text file`);
-              file_browser.isText = true;
-              file_browser.contents = hljs.highlightAuto(contents.toString('utf8')).value;
-              callback(null);
+      async () => {
+        const isBinary = await isBinaryFile(file_browser.paths.workingPath);
+        file_browser.isBinary = isBinary;
+        if (isBinary) {
+          const type = await FileType.fromFile(file_browser.paths.workingPath);
+          if (type) {
+            if (type?.mime.startsWith('image')) {
+              file_browser.isImage = true;
+            } else if (type?.mime === 'application/pdf') {
+              file_browser.isPDF = true;
             }
-          },
-          (err) => {
-            if (ERR(err, callback)) return;
-            callback(null); // should never get here
           }
-        );
+        } else {
+          // This is probably a text file. If it's is larger that 10MB, don't
+          // attempt to read it; treat it like an opaque binary file.
+          const { size } = await fs.stat(file_browser.paths.workingPath);
+          if (size > 10 * 1024 * 1024) {
```

Comment:
I picked this completely arbitrarily.

Reply:
I'd also be fine with something lower. Maybe 1 MB? That's still a huge text file.

Reply:
Good call - I originally had this as the first step of this function before we even know if we're dealing with a binary file or not, but it makes perfect sense to lower this now that we're reasonably confident that this is a text file!

Reply:
Lowered to 1MB in aad67da.